docs/architecture/grpc-for-wcf-developers/apm.md

```diff
@@ -1,11 +1,221 @@
+
+### How distributed tracing works
+
+Distributed tracing is based on the concept of *Spans*: named, timed operations that are part of a single *Trace*, which may involve processing on multiple nodes of a system. When a new operation is initiated a trace is created with a unique identifier. For each sub-operation, a span is created with its own identifier as well as the trace identifier. As the request passes around the system, various components can create *child* spans which include the identifier of their *parent*. A span has a *context*, which contains the trace and span identifiers, as well as useful data in the form of key/value pairs (called *baggage*).
+
+### Distributed tracing with DiagnosticSource
+
+.NET Core has an internal module that maps very well to distributed traces and spans: [DiagnosticSource](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/DiagnosticSourceUsersGuide.md). As well as providing a simple way to produce and consume diagnostics within a process, the DiagnosticSource module has the concept of an *Activity*, which is effectively an implementation of a distributed trace, or a span within a trace. The internals of the module take care of parent/child activities, including allocating identifiers.
+
+Because DiagnosticSource is a part of the core framework, it is supported by several core components, including `HttpClient`, Entity Framework Core and ASP.NET Core, including explicit support in the gRPC framework. When ASP.NET Core receives a request, it will check for a pair of HTTP headers matching the [W3C Trace Context](https://www.w3.org/TR/trace-context) standard. If the headers are found, an Activity is started using the identity values and context from the headers. If no headers are found, an Activity is started with generated identity values that match the standard format. Any diagnostics generated by the framework or by application code during the lifetime of this Activity can be tagged with the trace and span identifiers. The `HttpClient` support extends this further by checking for a current Activity on every request and automatically adding the trace headers to the outgoing request.
+
+The gRPC client and server libraries include explicit support for DiagnosticSource and Activity, and will create activities and apply and use header information automatically.
```

Comment:
Maybe worth clarifying that you're talking about grpc-dotnet here. Not all gRPC implementation do this out of the box.